extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -717,13 +716,32 @@ public List<SubtitlesStream> getSubtitles(final MediaFormat format) throws Parsi
                         .setMediaFormat(format)
                         .setLanguageCode(languageCode)
                         .setAutoGenerated(isAutoGenerated)
+                        .setAutoTranslated(false)
                         .build());
+                if (i == 0 && caption.getBoolean("isTranslatable")
```

Comment:
I would not base the extraction on the index, but rather on whether the subtitles are auto-generated:

```suggestion
                if (isAutoGenerated && caption.getBoolean("isTranslatable")
```

Also, this PR doesn't add support of subtitles translation for uploaded subtitles. For instance, see https://www.youtube.com/watch?v=_cMxraX_5RE: you can translate from German to French and from English to French, and the translations are different.

We may need another property in `SubtitlesStream` for this.

Reply:
I don't understand why we should use `isAutoGenerated` here. For better quality, it should be `!isAutoGenerated`. Manually added captions should be exact.
I was also wondering whether we should provide the auto-translated captions by default. Extracting the data for and generating ~100 SubtitleStreams takes some time. I'd definitely not recommend to do this for all available languages by default. On the other hand, we could provide a method which does this when needed.